tensorflow_datasets/scripts/metadata_cleanup.py

```diff
@@ -0,0 +1,92 @@
+# Lint as: python3
+"""Script to Cleanup Metadata directory.
+
+To test:
+Goto tensorflow_datasets/scripts and run using python metadata_cleanup.py
+
+```
+cd tensorflow_datasets/scripts
+python metadata_cleanup.py
+```
+It returns :
+
+Total number of registered datasets
+Total number of subdirectories in metadata direcotry
+Total number of meta directories removed 
+"""
+
+from absl import app
+import os
+import re
+import shutil
+import tensorflow as tf
+from tensorflow_datasets.core import registered
+
+metadata_path = os.path.join(os.pardir, "testing/metadata")
+
+def _extract_full_names_from_dirs(metadata_dir):
+    '''Get all metadata direcotry versions paths '''
+    meta_paths = set()
+    for root, dirs, files in os.walk(metadata_dir):

```

Comment:
You should not add all sub-directories. Only extract the `dataset_name/version` or `dataset_name/config/versions`.

Reply:
Yes this function is exactly doing this it takes only all `dataset_name/version` or `dataset_name/config/versions` from metadata folder, thats why I haven't include `fileName` in `meta_paths ` so it give only versions paths not files paths